script/stop-update.lua

```diff
@@ -59,12 +59,13 @@ function UpdateStop(stopID)
     global.LogisticTrainStops[stopID].parkedTrainID = nil
   end
 
-  -- remove invalid activeDeliveries -- shouldn't be necessary
-  -- for i=#stop.activeDeliveries, 1, -1 do
-    -- if not global.Dispatcher.Deliveries[stop.activeDeliveries[i]] then
-      -- table.remove(stop.activeDeliveries, i)
-    -- end
-  -- end
+  -- remove invalid activeDeliveries -- shouldn't be necessary -- only check oldest delivery

```

Comment:
Do your changes require this?
Why only check oldest? The 3rd of 5 deliveries can become invalid if the train doing it was destroyed.